src/connectors/autocomplete/__tests__/connectAutocomplete-test.js

```diff
@@ -167,4 +167,155 @@ See documentation: https://www.algolia.com/doc/api-reference/widgets/autocomplet
+      const makeWidget = connectAutocomplete(renderFn);
+      const widget = makeWidget({
+        indices: [
+          { label: 'Second', value: 'secondIndex' },
+          { label: 'Third', value: 'thirdIndex' },
+        ],
+      });
+
+      widget.init({ helper, instantSearchInstance: {} });
+
+      expect(detachDerivedHelper).toHaveBeenCalledTimes(0);
+
+      widget.dispose({ helper, state: helper.state });
+
+      expect(detachDerivedHelper).toHaveBeenCalledTimes(2);
+    });
+
+    it('removes only the DerivedHelper created by autocomplete', () => {
+      const detachDerivedHelper = jest.fn();
+      const helper = algoliasearchHelper(fakeClient, 'firstIndex');
+      helper.detachDerivedHelper = detachDerivedHelper;
+
+      const renderFn = () => {};
+      const makeWidget = connectAutocomplete(renderFn);
+      const widget = makeWidget({
+        indices: [
+          { label: 'Second', value: 'secondIndex' },
+          { label: 'Third', value: 'thirdIndex' },
+        ],
+      });
+
+      const derivedHelperOne = helper.derive(state => state);
+      const derivedHelperTwo = helper.derive(state => state);
+
+      widget.init({ helper, instantSearchInstance: {} });
+
+      expect(detachDerivedHelper).toHaveBeenCalledTimes(0);
+
+      widget.dispose({ helper, state: helper.state });
+
+      expect(detachDerivedHelper).toHaveBeenCalledTimes(2);
+      expect(helper.derivedHelpers).toEqual(
+        expect.arrayContaining([derivedHelperOne, derivedHelperTwo])
+      );
+    });
+
+    it('removes the `query` from the `SearchParameters`', () => {
+      const helper = algoliasearchHelper(fakeClient, 'firstIndex', {
+        query: 'Apple',
+      });
+
+      const renderFn = () => {};
+      const makeWidget = connectAutocomplete(renderFn);
+      const widget = makeWidget();
+
+      widget.init({ helper, instantSearchInstance: {} });
+
+      expect(helper.state.query).toBe('Apple');
+
+      const nextState = widget.dispose({ helper, state: helper.state });
+
+      expect(nextState.query).toBeUndefined();
+    });
+
+    it('removes the TAG_PLACEHOLDER from the `SearchParameters`', () => {
+      const helper = algoliasearchHelper(fakeClient, 'firstIndex', {
+        ...TAG_PLACEHOLDER,
+      });
+
+      const renderFn = () => {};
+      const makeWidget = connectAutocomplete(renderFn);
+      const widget = makeWidget();
+
+      expect(helper.state.highlightPreTag).toBe(
+        TAG_PLACEHOLDER.highlightPreTag
+      );
+
+      expect(helper.state.highlightPostTag).toBe(
+        TAG_PLACEHOLDER.highlightPostTag
+      );
+
+      widget.init({ helper, instantSearchInstance: {} });
+
+      const nextState = widget.dispose({ helper, state: helper.state });
+
+      expect(nextState.highlightPreTag).toBeUndefined();
+      expect(nextState.highlightPostTag).toBeUndefined();
+    });
+
+    it('removes the TAG_PLACEHOLDER from the `SearchParameters` only with `escapeHTML`', () => {
```

Comment:
```suggestion
    it('does not remove the TAG_PLACEHOLDER from the `SearchParameters` with `escapeHTML`', () => {
```

I had trouble understanding the test because of the name.